web/src/app/components/header/current-user-widget/current-user-widget.component.spec.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthService } from 'app/services/auth/auth.service';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CurrentUserWidgetComponent } from 'app/components/header/current-user-widget/current-user-widget.component';
+import { Subject } from 'rxjs';
+import { MatDialog } from '@angular/material/dialog';
+import { User } from 'app/shared/models/user.model';
+
+describe('CurrentUserWidgetComponent', () => {
+  let component: CurrentUserWidgetComponent;
+  let fixture: ComponentFixture<CurrentUserWidgetComponent>;
+  const user$ = new Subject<User | null>();
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      providers: [
+        { provide: AuthService, useValue: { user$, getUser$: () => user$ } },
+        { provide: MatDialog, useValue: {} },
+      ],
+      declarations: [CurrentUserWidgetComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CurrentUserWidgetComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```

Comment:
Can we add a test for `openProfileDialog` function?
(I realized this is not a new component, the function is extracted from other component, but you add a test, this component will reach 100% coverage :D)

Reply:
I plan to do an iteration over all components one I finish reshuffling them. Ok to submit the pure refactor first?